lib/Frontend/CompilerInvocation.cpp

```diff
@@ -111,6 +111,59 @@ static void updateRuntimeLibraryPaths(SearchPathOptions &SearchPathOpts,
   }
 }
 
+// FIXME: This function is a bit of a layering violation.  It should be
+// dispersed into IRGen option parsing if possible.
```

Comment:
It would be nice to explain specifically how it violates layering--would help a noobie. Also since it is required to preserve the incoming settings, where do they come from?